### PR TITLE
[VL] Make bloom_filter_agg fall back when might_contain is not transformable

### PR DESCRIPTION
**Pull request by zhli1142015**
## What changes were proposed in this pull request?
1. Reapply the change "[VL] Make bloom_filter_agg fall back when might_contain is not transformable".
2. Fix the issue only might_contain fallback.
3. Add flag to disable this in CK.
4. Add more UTs.

## How was this patch tested?

UT.

### CODE REVIEW

**Comment by zhli1142015** [comment]
cc @zhouyuan , @baibaichen , thanks.

**Comment by zhouyuan** [comment]
CC: @loneylee

**Comment by zhli1142015** [comment]
@zhouyuan , @baibaichen , are you ok to reapply this change?
@baibaichen and @loneylee , bloom_filter_agg fallback rule is disabled for CK so there is no impact to you. You can remove the flag `enableBloomFilterAggFallbackRule` after you confirm this change works ok for CK.
Thanks.

File: gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala [hunk 1]
```diff
@@ -269,6 +270,52 @@ case class FallbackEmptySchemaRelation() extends Rule[SparkPlan] {
...
+          TransformHints.tagNotTransformable(agg, "related BloomFilterMightContain falls back")
+          tagNotTransformableRecursive(agg.child)
+        case a: org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanExec =>
+          tagNotTransformableRecursive(a.executedPlan)
+        case _ =>
+          p.children.map(tagNotTransformableRecursive)
+      }
+    }
+
+    plan.transformExpressions {
```
Comment (zhouyuan) [inline gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala:311]: @zhli1142015 is this the key change? 

`Fix the issue only might_contain fallback.`

**Review by zhouyuan — APPROVED** [review]
+1

File: gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala (hunk 1 above)
Comment (zhli1142015) [reply gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala:311]: yes.

**Pull request merged**